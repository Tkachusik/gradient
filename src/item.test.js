import React from 'react';
import renderer from 'react-test-renderer';
import Item from './item';
import { BrowserRouter as Router } from 'react-router-dom';

describe("Test item component", () => {
    test("Should render component", () => {
        const item = { colorLeft: "#333", colorRight: "#222" };
        const component = renderer.create(<Router><Item id="1" item={item} deleteGradient={() => {}} /></Router>);
        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});