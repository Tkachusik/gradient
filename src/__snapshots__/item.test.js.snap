// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test item component Should render component 1`] = `
<div
  className="item"
>
  <a
    href="/edit/1"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "background": "linear-gradient(90deg, #333 0%, #222 100%)",
          "height": "50px",
        }
      }
    >
      <span
        className="text-left"
      >
        #333
      </span>
      <span
        className="text-right"
      >
        #222
      </span>
    </div>
  </a>
  <div
    className="delete-btn"
    onClick={[Function]}
  >
    <span>
      X
    </span>
  </div>
</div>
`;
